updater.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"bytes"
+	"crypto"
+	"encoding/hex"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/WiseGrowth/go-wisebot/logger"
+	update "github.com/inconshreveable/go-update"
+)
+
+// Updater is in charge to update the operator
+type Updater struct {
+	uploading bool
+	baseURL   string
+	version   string
+}
+
+// NewUpdate constructor method for Updater struct
+func NewUpdate(baseURL string, currentVersion string) *Updater {
+	return &Updater{uploading: false, baseURL: baseURL, version: currentVersion}
+}
+
+// Update method used for update this operator service based of param version
+func (up *Updater) Update(newVersion string) error {
+	//TODO: validate newVersion format, eg. 1.2.3
+	if newVersion == up.version {
+		err := errors.New("new version is the same that currently exists in the service")
+		return err
+	}
+
+	up.uploading = true
+
+	var urlChecksum = up.baseURL + newVersion + ".checksum"
+	var urlBin = up.baseURL + newVersion
+
+	log := logger.GetLogger()
+	log.Info("New Version: ", newVersion)
+	log.Info("URL Checksum: ", urlChecksum)
+	log.Info("URL Bin: ", urlBin)
+
+	//define http client
+	client := http.Client{}
```

Comment:
I think that client must be a point to a http.Client instead a simple instance. And, if your client has no custom configs of specific parameter, maybe  you should try with `client := http.DefaultClient`

Reply:
lo intentamos pero no funcionó

Reply:
read the compiler messages